# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/CoalescedTouchData.cpp
# Commit: f8a2a169b296
# Full Hash: f8a2a169b2969d29e04733ae27a56459e5359973
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-05-16 09:19:19
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Revert "Bug 1550462 - part 3: Enable `pointerrawupdate` event r=smaug" for causing mochitest failures @ test_pointerrawupdate_event_count.html
#   
#   This reverts commit eb27445e0352d6b4df540adb40e21f72fc052e80.
#   
#   Revert "Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar"
# ==============================================================================

diff -r 8d25773fc82f -r f8a2a169b296 dom/ipc/CoalescedTouchData.cpp
--- a/dom/ipc/CoalescedTouchData.cpp	Fri May 16 06:20:47 2025 +0000
+++ b/dom/ipc/CoalescedTouchData.cpp	Fri May 16 09:30:09 2025 +0300
@@ -28,7 +28,6 @@
     PointerEventHandler::InitPointerEventFromTouch(*event, aEvent, *touch);
     event->mFlags.mBubbles = false;
     event->mFlags.mCancelable = false;
-    event->convertToPointerRawUpdate = false;
   }
 }
 
@@ -48,9 +47,10 @@
     MOZ_ASSERT(mCoalescedInputEvent->mModifiers == aEvent.mModifiers);
     MOZ_ASSERT(mCoalescedInputEvent->mInputSource == aEvent.mInputSource);
 
-    for (const RefPtr<Touch>& touch : aEvent.mTouches) {
+    for (size_t i = 0; i < aEvent.mTouches.Length(); i++) {
+      const RefPtr<Touch>& touch = aEvent.mTouches[i];
       // Get the same touch in the original event
-      const RefPtr<Touch> sameTouch = GetTouch(touch->Identifier());
+      RefPtr<Touch> sameTouch = GetTouch(touch->Identifier());
       // The checks in CoalescedTouchData::CanCoalesce ensure it should never
       // be null.
       MOZ_ASSERT(sameTouch);
@@ -58,7 +58,6 @@
       MOZ_ASSERT(!sameTouch->mCoalescedWidgetEvents->mEvents.IsEmpty());
       if (!sameTouch->Equals(touch)) {
         sameTouch->SetSameAs(touch);
-        sameTouch->convertToPointerRawUpdate = touch->convertToPointerRawUpdate;
         WidgetPointerEvent* event =
             sameTouch->mCoalescedWidgetEvents->mEvents.AppendElement(
                 WidgetPointerEvent(aEvent.IsTrusted(), ePointerMove,
@@ -73,18 +72,6 @@
   }
 }
 
-void CoalescedTouchData::NotifyTouchRawUpdateOfHandled(
-    const WidgetTouchEvent& aEvent) {
-  if (IsEmpty()) {
-    return;
-  }
-  for (const RefPtr<Touch>& touch : aEvent.mTouches) {
-    if (const RefPtr<Touch> sameTouch = GetTouch(touch->Identifier())) {
-      sameTouch->convertToPointerRawUpdate = false;
-    }
-  }
-}
-
 bool CoalescedTouchData::CanCoalesce(const WidgetTouchEvent& aEvent,
                                      const ScrollableLayerGuid& aGuid,
                                      const uint64_t& aInputBlockId,